arroyo-compiler-service/src/main.rs

```diff
@@ -209,6 +209,13 @@ impl CompileService {
             }
         }
 
+        // clean up the udfs_dir
+        if !self.debug {
+            tokio::fs::remove_dir_all(&self.build_dir.join("udfs_dir"))
+                .await
+                .unwrap();
```

Comment:
These should return errors instead of unwrapping so that errors are properly propagated to the client